File: projet/client_leger/polyquiz/lib/models/host_interface.dart
import 'package:polyquiz/models/quiz.dart';

class InitialQuestionData {
  final QuizQuestion question;
  final String username;
  final int index;
  final int numberOfQuestions;

  InitialQuestionData({
    required this.question,
    required this.username,
    required this.index,
    required this.numberOfQuestions,
  });
}

class NextQuestionData {
  final QuizQuestion question;
  final int index;
  final bool isLast;

  NextQuestionData({
    required this.question,
    required this.index,
    required this.isLast,
  });
}

File: projet/common/interfaces/host.interface.ts
import { QuizQuestion } from './quiz.interface';
import {Player} from "../constants/player-list.component.const";

export interface InitialQuestionData {
    question: QuizQuestion;
    username: string;
    index: number;
    numberOfQuestions: number;
}

export interface NextQuestionData {
    question: QuizQuestion;
    index: number;
    isLast: boolean;
}

export interface ObsQuestionData {
    question: QuizQuestion;
    username: string;
    index: number;
    numberOfQuestions: number;
    isLast: boolean;
    gameActivityStatus: [number, number];
}


export interface HostCurrentGameInterface {
    roomId: number;
    timerText: string;
    isGameOver: boolean;
    leftPlayers: Player[];
    players: Player[];
    histogramDataChangingResponses: [number, number] | number[];
    isHostEvaluating: boolean;
    gameStats: string;
    isPaused: boolean;
    isPanicMode: boolean;
}
